src/main/java/com/mentoria/helena/confeitaria/repository/FuncionarioRepository.java

```diff
@@ -0,0 +1,36 @@
+package com.mentoria.helena.confeitaria.repository;
+
+import com.mentoria.helena.confeitaria.classes.Cliente;
+import com.mentoria.helena.confeitaria.classes.Funcionario;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class FuncionarioRepository implements IFuncionarioRepository{
+    ArrayList<Funcionario> listaFuncionario = new ArrayList();
```

Comment:
Quando a gente declara uma variável de uma classe que na verdade é uma classe herdada de outra abstrata, é sempre recomendado usar o tipo abstrato ao invés do concreto. Em outras palavras, `ArrayList` é o tipo concreto, que implementa a classe `List`. Então quando declarar a variável, declara ela como sendo do tipo `List` para que isso fique genérico para quem for usar depois.

Reply:
apesar que nesse caso podemos remover essa variável daqui já que é usado o HashMap não?